src/datasources/asset/components/AssetQueryCalibrationForecastEditor.tsx

```diff
@@ -0,0 +1,57 @@
+import { SelectableValue, toOption } from '@grafana/data';
+import { AssetDataSource } from '../AssetDataSource';
+import { AssetCalibrationForecastGroupByType, AssetCalibrationForecastQuery, AssetQuery } from '../types';
+import { InlineField, MultiSelect } from '@grafana/ui';
+import React from 'react';
+import { enumToOptions } from '../../../core/utils';
+import _ from 'lodash';
+
+type Props = {
+  query: AssetCalibrationForecastQuery;
+  handleQueryChange: (value: AssetQuery, runQuery: boolean) => void;
+  datasource: AssetDataSource;
+};
+
+export function QueryCalibrationForecastEditor({ query, handleQueryChange, datasource }: Props) {
+  query = datasource.prepareQuery(query) as AssetCalibrationForecastQuery;
+  const handleGroupByChange = (items?: Array<SelectableValue<string>>): void => {
+    if (!items || _.isEqual(query.groupBy, items)) {
+      return;
+    }
+
+    let groupBy = [];
+    let locationIndex = items.findIndex(item => item.value === AssetCalibrationForecastGroupByType.Location);
+    if (locationIndex !== -1) {
+      groupBy.push(AssetCalibrationForecastGroupByType.Location);
+      items.splice(locationIndex, 1);
+    }
+
+    if (items.length) {
+      groupBy.push(items[items.length - 1].value!);
```

Comment:
items can have maximum 2 elements, and you already added the location (if present), and here you add just the time-based grouping element, right? if so, then  items.length - 1 is always 0